force-app/main/default/objects/Lead/fields/Birth_date__c.field-meta.xml

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>Birth_date__c</fullName>
```

Comment:
Casing issue. You may have to rename all of the files. Casing changes for file names can be challenging to handle with git. I suggest using git mv
```suggestion
    <fullName>Birth_Date__c</fullName>
```